### PR TITLE
feat(x/ecocredit)!: migrate add-allow-denom proposal to msg based proposal

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: proto/regen/ecocredit/marketplace/v1/types.proto [hunk 1]
```diff
@@ -22,4 +19,4 @@ message AllowDenomProposal {
   // denom contains coin denom information that will be added to the
   // list of allowed denoms for use in the marketplace.
   AllowedDenom denom = 3;
-}
\ No newline at end of file
+}
```
Comment (aleem1314) [inline proto/regen/ecocredit/marketplace/v1/types.proto:22]:
```suggestion
} 
```

File: proto/regen/ecocredit/marketplace/v1/types.proto [hunk 2]
```diff
@@ -22,4 +19,4 @@ message AllowDenomProposal {
...
```
Comment (aleem1314) [inline proto/regen/ecocredit/marketplace/v1/types.proto:22]:
```suggestion
}
```

File: proto/regen/ecocredit/marketplace/v1/types.proto [hunk 3]
```diff
@@ -5,9 +5,6 @@ package regen.ecocredit.marketplace.v1;
 import "gogoproto/gogo.proto";
 import "regen/ecocredit/marketplace/v1/state.proto";
 
-option go_package =
```
Comment (aleem1314) [inline proto/regen/ecocredit/marketplace/v1/types.proto:8]: We are not using `AllowDenomProposal` message. Removing this line is breaking `buf-lint` check.

File: x/ecocredit/server/marketplace/features/msg_add_allowed_denom.feature [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        "bank_denom":"uregen",
+        "display_denom":"REGEN",
+        "exponent":6
+      }
+      """
+      Then expect error contains "expected gov account as only signer for proposal message"
+
+  Rule: The allowed denom does not exists
+
+    Scenario: The allowed denom does not exists
```
Comment (technicallyty) [inline x/ecocredit/server/marketplace/features/msg_add_allowed_denom.feature:36]:
```suggestion
    Scenario: The allowed denom does not exist
```

**Review by technicallyty — APPROVED** [review]
lgtm, 2 tiny nits

File: x/ecocredit/server/marketplace/msg_add_allowed_denom_test.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	api "github.com/regen-network/regen-ledger/api/regen/ecocredit/marketplace/v1"
+	"github.com/regen-network/regen-ledger/x/ecocredit/marketplace"
+)
+
+type addAllowedDenomSuite struct {
+	*baseSuite
+	err error
+}
+
+func TestAddCreditType(t *testing.T) {
```
Comment (aleem1314) [inline x/ecocredit/server/marketplace/msg_add_allowed_denom_test.go:19]:
```suggestion
func TestAddAllowedDenom(t *testing.T) {
```

**Review by ryanchristo — APPROVED** [review]
[tACK](https://hackmd.io/h_irULsYQdKdCtRH31mjLA)

**Comment by ryanchristo** [comment]
Each file within a proto package must have the same go package option. Even though we don't need the proto for the proposal in marketplace types, we need to keep the go package option.

```
Files in package "regen.ecocredit.marketplace.v1" have both values "github.com/regen-network/regen-ledger/x/ecocredit/marketplace" and no value for option "go_package" and all values must be equal.
```

**Pull request merged**